nixos/modules/services/games/minecraft-server.nix

```diff
@@ -172,16 +181,39 @@ in {
     };
     users.groups.minecraft = {};
 
+    systemd.sockets.minecraft-server = {
+      bindsTo = [ "minecraft-server.service" ];
+      socketConfig = {
+        ListenFIFO = "/run/minecraft-server.stdin";
+        SocketMode = "0660";
+        SocketUser = "minecraft";
+        SocketGroup = "minecraft";
+        RemoveOnStop = true;
+        FlushPending = true;
+      };
+    };
+
     systemd.services.minecraft-server = {
       description   = "Minecraft Server Service";
       wantedBy      = [ "multi-user.target" ];
-      after         = [ "network.target" ];
+      requires      = [ "minecraft-server.socket" ];
+      after         = [ "network.target" "minecraft-server.socket" ];
 
       serviceConfig = {
         ExecStart = "${cfg.package}/bin/minecraft-server ${cfg.jvmOpts}";
+        ExecStop = "${stopScript} $MAINPID";
         Restart = "always";
         User = "minecraft";
         WorkingDirectory = cfg.dataDir;
+
+        KillSignal = "SIGCONT";
+        TimeoutStopSec = "120s";
+        RestartSec = "15s";
```

Comment:
Can you please explain why these values were added? My understanding is that:

1. Telling systemd to send `SIGCONT` is basically making sure it never kills the process on its own (since, if my understanding of this is correct, `SIGCONT` is a noop if the process is running), and waits for the stop script to complete
2. This makes sure that `SIGCONT` isn't sent too early(?)
3. Why add a higher wait in between (automatic) restarts?

Thanks!

Reply:
I dont fully know what forge mods do exactly, but some get really really mad if you start them up inside the same second that it just shut down on. So a `RestartSec = "2s"` may have worked just as well, but i have just pushed it up in `5s` increments each time i had issues with a Forge modpack complaining about starting up too soon from its last shutdown.

For the exacts, i sadly cant say, it was 2 years ago i put these defaults into my systemd service file. But they are not needed, so thanks for spotting that.